crossfit/backend/torch/hf/model.py

```diff
@@ -19,24 +19,67 @@
 import joblib
 import numpy as np
 import torch
-from sklearn.linear_model import LinearRegression
-from tqdm import tqdm
 from transformers import AutoConfig, AutoModel, AutoTokenizer
 
+from crossfit.backend.torch.hf.memory_curve_utils import fit_memory_estimate_curve
 from crossfit.backend.torch.model import Model
 from crossfit.dataset.home import CF_HOME
-from crossfit.utils.model_adapter import adapt_model_input
 
 
 class HFModel(Model):
-    def __init__(self, path_or_name: str, max_mem_gb: int = 16, training=False):
+    def __init__(
+        self,
+        path_or_name: str,
+        max_mem_gb: int = 16,
+        training: bool = False,
+        start_batch_size: int = 1,
+        end_batch_size: int = 2048,
+        batch_size_increment: int = 256,
+        start_seq_len: int = 1,
+        seq_len_increment: int = 64,
+    ):
         super().__init__(path_or_name, max_mem_gb)
-
-        if not training:
-            with torch.no_grad():
-                self.fit_memory_estimate_curve()
+        self.start_batch_size = start_batch_size
+        self.end_batch_size = end_batch_size
+        self.batch_size_increment = batch_size_increment
+        self.start_seq_len = start_seq_len
+        self.seq_len_increment = seq_len_increment
+
+        cache_dir = os.path.join(
+            CF_HOME, "memory", AutoConfig.from_pretrained(path_or_name)._name_or_path
```

Comment:
Looks like a small bug was introduced here. This used to call `self.load_cfg` instead of using `AutoConfig`, and it should go back to calling it. During my latest tests this is causing an error because for my config I need to pass in a user access token.

Reply:
Resolved by  https://github.com/rapidsai/crossfit/pull/66/commits/15e27bb33c8a2d56fc8c165d45218c8c902424c0 and https://github.com/rapidsai/crossfit/pull/66/commits/15e27bb33c8a2d56fc8c165d45218c8c902424c0